frontend/lib/src/components/elements/DeckGlJsonChart/DeckGlJsonChart.test.tsx

```diff
@@ -62,84 +69,151 @@ const getProps = (
 
   return {
     element: DeckGlJsonChartProto.create({
+      id: mockId,
       json: JSON.stringify(json),
       ...elementProps,
     }),
     width: 0,
     mapboxToken: "mapboxToken",
     height: undefined,
     theme: mockTheme.emotion,
+    isFullScreen: false,
   }
 }
 
 describe("DeckGlJsonChart element", () => {
   it("renders without crashing", () => {
     const props = getProps()
-    const wrapper = shallow(<DeckGlJsonChart {...props} />)
 
-    expect(wrapper.find(DeckGL).length).toBe(1)
-    expect(wrapper.find(NavigationControl).length).toBe(1)
+    render(<DeckGlJsonChart {...props} />)
+
+    const deckGlJsonChart = screen.getByTestId("stDeckGlJsonChart")
+    expect(deckGlJsonChart).toBeInTheDocument()
   })
 
-  it("merges client and server changes", () => {
+  it("should merge client and server changes in viewState", () => {
     const props = getProps()
-    const wrapper = shallow(<DeckGlJsonChart {...props} />)
-    const deckGL = wrapper.find(DeckGL)
+    const initialViewStateServer = mockInitialViewState
 
-    expect(deckGL.length).toBe(1)
-    expect(deckGL.prop("onViewStateChange")).toBeDefined()
+    const initialViewStateClient = { ...mockInitialViewState, zoom: 8 }
 
-    // @ts-expect-error
-    deckGL.prop("onViewStateChange")({
-      viewState: { pitch: 5, zoom: 5 },
-    })
+    const state = {
```

Comment:
Changed this test to not involve calling DeckGl methods because that seems odd. So I just changed this test to test our `getDerivedStateFromProps` function. 

Testing the changing of pydeck with unit testing and RTL is turning out to be a little too difficult as we can click the button but it's not immediately obvious how to check the map changed. That should likely be tested with e2e tests and thus changing this test seems like a reasonable thing from my perspective.